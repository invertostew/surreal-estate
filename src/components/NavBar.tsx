import React from "react";

import "../styles/NavBar.css";

function NavBar() {
  return (
    <nav className="navbar">
      <ul className="navbar-links">
        <li className="navbar-links__item">
          <a href="#!">View Properties</a>
        </li>
        <li className="navbar-links__item">
          <a href="#!">Add a Property</a>
        </li>
      </ul>
    </nav>
  );
}

export default NavBar;
