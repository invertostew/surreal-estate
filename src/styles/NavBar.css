.navbar-links {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.navbar-links__item {
  display: inline-block;
  margin-right: 0.5rem;
  padding: 0.5rem;
}

.navbar-links__item:first-of-type {
  margin-left: 0.5rem;
}

.navbar-links__item:last-of-type {
  margin-right: 0;
}

.navbar-link {
  text-decoration: none;
  color: #3a3b3c;
}

.navbar-link.selected {
  color: #1d4ed8;
}
