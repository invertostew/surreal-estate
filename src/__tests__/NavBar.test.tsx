import React from "react";
import { render, screen } from "@testing-library/react";

import NavBar from "../components/NavBar";

describe("NavBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<NavBar />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the navbar links", () => {
    render(<NavBar />);

    const viewPropertiesLink = screen.getByText(/view properties/i);
    const addPropertyLink = screen.getByText(/add a property/i);

    expect(viewPropertiesLink).toBeInTheDocument();
    expect(addPropertyLink).toBeInTheDocument();
  });
});
