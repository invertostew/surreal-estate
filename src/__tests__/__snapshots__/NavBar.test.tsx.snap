// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links__item"
      >
        <a
          href="#!"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links__item"
      >
        <a
          href="#!"
        >
          Add a Property
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
